### PR TITLE
using request, session, global pid to select patient

### PR DESCRIPTION
**Pull request by sloboy**
add_edit_event only used the request to gather pid.   This changes it to use session if no request pid, and global $pid if no session.  Sets no pid if all three fail.  Thanks @sjpadgett for confirming bug and suggested fix.  I'm still new to this:)

### CODE REVIEW

File: interface/main/calendar/add_edit_event.php [hunk 1]
```diff
@@ -874,6 +874,10 @@ function setEventDate($start_date, $recurrence)
     $patientid = '';
     if ($_REQUEST['patientid']) {
         $patientid = $_REQUEST['patientid'];
+    } elseif (isset($_SESSION['pid'])) {
+        $patientid = ($_SESSION['pid']);
+    } elseif ($pid>0){
```
Comment (sjpadgett) [inline interface/main/calendar/add_edit_event.php]: For clarity i'd use either $pid or $_SESSION as both are not needed. $pid is set from calling globals which essentially comes from the session super. Also, I wouldn't use isset() to test the session super but instead use !empty($_SESSION['pid']). Lastly, $pid is best typed to an int when doing conditional testing i.e $pid + 0 > 0.

Comment (sloboy) [reply interface/main/calendar/add_edit_event.php]: Thank you for the suggestions!  I'm a longtime programmer (all the way back to logo and pascal on an apple 2e) but relatively new to php and git.  I welcome the tips because I am learning as I go.

Comment (sjpadgett) [reply interface/main/calendar/add_edit_event.php]: Yep, I get it. Same with me with moving to a scripting language, just a different animal. Brady helped me a lot when I started out on OpenEMR so now that you've got the hang of it, go to town.

**Comment by sjpadgett** [comment]
Thanks for this. I'm surprised it got this far without notice. Bringing in.
